pyproject.toml

```diff
@@ -9,9 +9,10 @@ authors = [
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = "^3.11"
+python = "^3.12"
```

Comment:
Sounds good on keeping Python 3.12, we'll need to update the docker images and GitHub CI to 3.12 in this PR.

Docker images - we'll need to bump the versions on L6, L28, and L72. https://github.com/NOAA-GSL/VxIngest/blob/e1e82b5d8bcebca961b0161899b5c38b3228fc39/docker/ingest/Dockerfile#L6

GitHub CI - We'll need to update mentions of 3.11 in the lint & test jobs. E.g. - the lint job: https://github.com/NOAA-GSL/VxIngest/blob/e1e82b5d8bcebca961b0161899b5c38b3228fc39/.github/workflows/ci.yml#L24-L29

Reply:
Isn't this https://hub.docker.com/layers/library/python/3.12.4-slim-bookworm/images/sha256-48ba73b97df2ad08bfba9e954ceeddedc2e90aed387ce48ba561b0a6bccd3164 telling me that 3.12.4-slim-bookworm has two high vulnerabilities?

Reply:
I'm looking at the ingest Dockerfile L28

Reply:
I'd recommend targeting `FROM 3.12-slim-bookworm AS <layer>` so we get point updates. 

It does look like `pip` is showing a [high vulnerability in that version.](https://hub.docker.com/layers/library/python/3.12-slim-bullseye/images/sha256-9a8f510466b54509a8e7d9d7bc300401d85537d5a08ed9131e389fcba50234b3?context=explore) However, we update `pip` in our image (or at least, we will once https://github.com/NOAA-GSL/VxIngest/pull/403 is merged) so should pull in a patch if one exists. 

There's a chance this is also showing unpatched vulnerabilities, which GSL ignores since by definition we can't fix them.

Reply:
certainly